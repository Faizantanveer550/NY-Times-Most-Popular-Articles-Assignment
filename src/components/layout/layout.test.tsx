import { render, screen } from "@testing-library/react";
import Layout from "./layout";

test("Snapshot Test", () => {
  const children = <p>This is the content.</p>;
  const { asFragment } = render(<Layout>{children}</Layout>);
  expect(asFragment()).toMatchSnapshot();
});

test("Layout renders its children", () => {
  const children = <p>This is the content.</p>;
  render(<Layout>{children}</Layout>);

  expect(screen.getByText("This is the content.")).toBeInTheDocument();
});

test("Layout renders an empty container with no children", () => {
  render(<Layout />);

  expect(screen.getByTestId("layout-with-out-child")).toBeInTheDocument(); // Assuming LayoutStyled uses 'region' for layout
});
