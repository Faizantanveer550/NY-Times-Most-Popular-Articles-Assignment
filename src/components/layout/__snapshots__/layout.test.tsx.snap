// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Test 1`] = `
<DocumentFragment>
  <div
    class="sc-aYaIB knrHSB"
  >
    <p>
      This is the content.
    </p>
  </div>
</DocumentFragment>
`;
